### PR TITLE
Implement correct behavior of multiple retrievals of pipeline state from libraries.

### PR DESCRIPTION
**Pull request by HansKristian-Work**
Both MHR and Elden Ring retrieve the same pipelines multiple times. The D3D12 pipeline library API is designed so that multiple retrievals is very fast and reuses an existing internally referenced state object. Avoids us having to validate, decompress, create SPIR-V objects, create pipeline cache blob, create pipeline, etc.

We still have to perform light validation, but pales in comparison to what we did ...

Also adds extensive test for the edge cases in ref-counting shenanigans.

### CODE REVIEW

File: libs/vkd3d/state.c [hunk 1]
```diff
@@ -1874,6 +1880,17 @@ static ULONG STDMETHODCALLTYPE d3d12_pipeline_state_Release(ID3D12PipelineState
 
         d3d12_device_release(device);
```
Comment (doitsujin) [inline libs/vkd3d/state.c:1881]: Do we test the device ref count behaviour? Normally we're supposed to release the device when the **public** ref count reaches 0, and AddRef the device when the public ref count goes from 0 to 1.

Probably not the end of the world here since the pipeline objects are owned by the pipeline library rather than the device itself, but generally this is done to avoid circular references.

Comment (HansKristian-Work) [reply libs/vkd3d/state.c:1881]: I can test this.

Comment (HansKristian-Work) [reply libs/vkd3d/state.c]: Correct, we have to release the device reference, but not the private data. Just d3d12 memes ...

**Review by doitsujin — APPROVED** [review]
